### ISSUE TITLE
Company completion very slow in scala mode

### ISSUE THREAD

**Issue by samspills** [syl20bnr/spacemacs#7584]
#### Description :octocat:

Completion with company is very slow in scala mode
#### Reproduction guide :beetle:
- Start Emacs (in terminal)
- Open a scala file
- type `val thing = otherThing.map(stuff)` without waiting for any completion

_Observed behaviour:_ :eyes: :broken_heart:
`.map(stuff)` takes an unreasonably long time to show up. If I'm trying to chain a bunch of operations together, it can be especially painful to wait for everything to populate. (Just to clarify, this is the only input lag that I experience)

_Expected behaviour:_ :heart: :smile:
Less (or no) delay. (Is there a way I can explicitly trigger completion, or can I add a wait time where company is only called if I pause for a half second?)
#### System Info :computer:
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.200.4
- Spacemacs branch: master (rev. 6feb70d)
- Graphic display: nil
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
  `elisp
  (better-defaults emacs-lisp git org markdown syntax-checking python scala)
  `

**Comment by TheBB** [comment]
> (Is there a way I can explicitly trigger completion, or can I add a wait time where company is only called if I pause for a half second?)

See `company-idle-delay` and `company-complete`

**Comment by TheBB** [comment]
Are you sure the delay is due to Emacs and not external completion tools (as is the case with clang)?

**Comment by samspills** [comment]
I did try setting company-idle-delay with a scala-mode-hook, but it didn't have an affect on this particular issue. I think ensime or scala mode has an auto-completion hook after the period, so the delay itself didn't matter.

After more experimenting, I have found that the delay does not occur every completion, but when there is a delay it's accompanied by these messages: 
```
 Company: An error occurred in auto-begin
 Company: backend ensime-company async timeout with args (candidates )
 Company: An error occurred in auto-begin
 Company: backend nil error "Symbol's function definition is void: nil" with args (annotation ++:)
```

Based on that, this does seem to be an issue with the Ensime backend timing out, and not necessarily an emacs issue.
